keras_nlp/layers/positional_embedding_test.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for Keras-based positional embedding layer."""
+
+import numpy as np
+import tensorflow as tf
+
+from keras_nlp.layers import positional_embedding
+
+
+class PositionalEmbeddingLayerTest(tf.test.TestCase):
```

Comment:
I think we also need these tests:

1. A full forward-backward pass that verifies the gradients can be propagated to the embedding layer.
2. Saving/loading that verifies the embedding is correctly restored.

Reply:
Could we also add a test that overrides the initializer with a callable?
https://keras.io/api/layers/initializers/#simple-callables

That would allow us to effectively write a correctness test for the whole layer I think.

```
def custom_init(shape, dtype=None):
    return tf.tile(tf.range(...)) # tile an index for each position

layer = PositionalEmbedding(max_length=32, initializer=custom_init)
```

Reply:
Adding all three plus a test for `get_config` / `from_config`